extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/groupby/GroupByQueryKit.java

```diff
@@ -62,6 +65,25 @@ public GroupByQueryKit(ObjectMapper jsonMapper)
     this.jsonMapper = jsonMapper;
   }
 
+  /**
+   * Intermediate signature of a particular {@link GroupByQuery}. Does not include post-aggregators, and all
+   * aggregations are nonfinalized.
+   */
+  private static RowSignature computeIntermediateSignature(final GroupByQuery query)
+  {
+    final RowSignature postAggregationSignature = query.getResultRowSignature(RowSignature.Finalization.NO);
+    final RowSignature.Builder builder = RowSignature.builder();
+
+    for (int i = 0; i < query.getResultRowSizeWithoutPostAggregators(); i++) {
+      builder.add(
+          postAggregationSignature.getColumnName(i),
+          postAggregationSignature.getColumnType(i).orElse(null)
+      );
+    }
```

Comment:
This assumes that the post aggregations are at the end of the result signature. I don't think there's a contract which has that. I think something more reliable could be: 
```java
for (col : rowSignature) {
  if (colName not in query.getPostAggregators()) {
    builder.add()
  }
}
```

Reply:
I don't know how this code was moved up. This was an existing function. Moved it back